### PR TITLE
Add a description about new data mapper code action

### PR DESCRIPTION
**Pull request by Ayodhya94**
## Purpose
Add a description about new automatic data mapping code action.

## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

File: swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md [hunk 1]
```diff
@@ -52,11 +52,19 @@ These allow you to perform the below tasks easily based on the diagnostics and t
...
 - Create variable definitions
 - Create an undefined function
+- Map record types automatically
 
 For example, you can add documentation for a function as shown below.
 
  ![Code actions](/swan-lake/learn/images/code-actions.gif)
 
+You can use automatic data mapping code action as shown below.
+ > **Note**: By default, the Automatic data mapping code action is disabled. In order to enable it, go to **Settings** and Select **Ballerina** from **Extensions**. There you can check the box in front of **Data Mapper: Enabled**.
```
Comment (praneesha) [inline swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md:62]:
```suggestion
 > **Note**: By default, the automatic data mapping code action is disabled. To enable it, click **Settings**, select **Ballerina** from **Extensions**, and select the **Data Mapper: Enabled** checkbox. (If you are using MacOS, you can go to **Settings** by pressing `Cmd` + `,`.)
```

File: swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md [hunk 2]
```diff
@@ -52,11 +52,19 @@ These allow you to perform the below tasks easily based on the diagnostics and t
...
 For example, you can add documentation for a function as shown below.
 
  ![Code actions](/swan-lake/learn/images/code-actions.gif)
 
+You can use automatic data mapping code action as shown below.
+ > **Note**: By default, the Automatic data mapping code action is disabled. In order to enable it, go to **Settings** and Select **Ballerina** from **Extensions**. There you can check the box in front of **Data Mapper: Enabled**. 
+
+ ![Code actions](/swan-lake/learn/images/data-mapper-code-action.gif)
+
+ > **Tip**: If you are using MacOS, you can go to **Settings** by pressing `Cmd` + `,`.
```
Comment (praneesha) [inline swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md:66]:
```suggestion
 > **Tip**: If you are using MacOS, you can go to **Settings** by pressing `Cmd` + `,`.
```
This tip can be removed.

File: swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md [hunk 3]
```diff
@@ -52,11 +52,19 @@ These allow you to perform the below tasks easily based on the diagnostics and t
 - Add missing imports 
 - Create variable definitions
 - Create an undefined function
+- Map record types automatically
 
 For example, you can add documentation for a function as shown below.
```
Comment (praneesha) [inline swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md:57]:
```suggestion
For another example, you can add documentation for a function as shown below.
```
Let's move this to place it after the data mapping code action content.

File: swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md [hunk 4]
```diff
@@ -52,11 +52,19 @@ These allow you to perform the below tasks easily based on the diagnostics and t
...
 
  ![Code actions](/swan-lake/learn/images/code-actions.gif)
 
+You can use automatic data mapping code action as shown below.
```
Comment (praneesha) [inline swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md:61]:
```suggestion
For example, you can use the automatic data mapping code action as shown below.
```

File: swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md [hunk 5]
```diff
@@ -52,11 +52,19 @@ These allow you to perform the below tasks easily based on the diagnostics and t
 - Add missing imports 
 - Create variable definitions
 - Create an undefined function
+- Map record types automatically
```
Comment (anupama-pathirage) [inline swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md:55]:
```suggestion
- Map the record types automatically
```

File: swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md [hunk 6]
```diff
@@ -52,11 +52,19 @@ These allow you to perform the below tasks easily based on the diagnostics and t
...
 
-For example, you can add documentation for a function as shown below.
+For another example, you can add documentation for a function as shown below.
 
  ![Code actions](/swan-lake/learn/images/code-actions.gif)
 
+For example, you can use the automatic data mapping code action as shown below.
+ > **Note**: By default, the automatic data mapping code action is disabled. To enable it, click **Settings**, select **Ballerina** from **Extensions**, and select the **Data Mapper: Enabled** checkbox. (If you are using MacOS, you can go to **Settings** by pressing `Cmd` + `,`.)
```
Comment (anupama-pathirage) [inline swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md:62]: Why do we need to repeat the sentence ` (If you are using MacOS, you can go to **Settings** by pressing `Cmd` + `,`.` here and as a `Tip` below.

File: swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md (hunk 5 above)
Comment (praneesha) [reply swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md:55]: It is fine to say "Map record types" here as we aren't specifying a particular record type but referring to record types in general.

File: swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md (hunk 6 above)
Comment (praneesha) [reply swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md:62]: The review comment was to remove the tip and add this in brackets.

File: swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md (hunk 2 above)
Comment (Ayodhya94) [reply swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md]: Removed this tip

File: swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md (hunk 6 above)
Comment (Ayodhya94) [reply swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md:58]: With the new commit the repeated sentence was removed

File: swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md [hunk 7]
```diff
@@ -52,8 +52,14 @@ These allow you to perform the below tasks easily based on the diagnostics and t
 - Add missing imports 
 - Create variable definitions
 - Create an undefined function
+- Map the record types automatically
 
-For example, you can add documentation for a function as shown below.
+For example, you can use the automatic data mapping code action as shown below.
+ > **Note**: By default, the automatic data mapping code action is disabled. To enable it, click **Settings**, select **Ballerina** from **Extensions**, and select the **Data Mapper: Enabled** checkbox. (If you are using MacOS, you can go to **Settings** by pressing `Cmd` + `,`.)
```
Comment (praneesha) [inline swan-lake/learn/tools-ides/setting-up-visual-studio-code/language-intelligence.md]:
```suggestion
 > **Note**: By default, the automatic data mapping code action is disabled. To enable it, click **Settings**, select **Ballerina** from **Extensions**, and select the **Data Mapper: Enabled** checkbox. (If you are using macOS, you can go to **Settings** by pressing `Cmd` + `,`.)
```

**Pull request merged**